### PR TITLE
[BUG] Fixed Umzug implementation & JWT Authentication issues

### PR DESCRIPTION
**Pull request by Redtigercod4**
Bugs fixed:
1. Fixed umzug bug for migrations so that new migrations can be added. 

2. Fixed Authentication issues with JWT Generation to allow ID Token & Access Token to be sent via Login API endpoint.

### CODE REVIEW

File: api/src/controllers/AuthenticationController.ts [hunk 1]
```diff
@@ -1,56 +1,58 @@
...
+
+export default class AuthenticationContoller {
   private authService = new AuthService();
 
-  public async signin(req: Request, res: Response) {
+  public async signIn(req: Request, res: Response) {
     try {
-      const data: UserAttributes = req.body;
-
+      const data: UserLogin = req.body;
```
Comment (SolarFactories) [inline api/src/controllers/AuthenticationController.ts:23]: S: Validate format of `req.body`, before annotating type from `any`.

File: api/src/controllers/AuthenticationController.ts [hunk 2]
```diff
@@ -1,56 +1,58 @@
...
-        accessToken: token,
-        userId: userDetails.id,
-        email: userDetails.email,
-        firstName: userDetails.firstName,
-        lastName: userDetails.lastName,
+        idToken: this.authService.generateIdToken(userDetails),
+        accessToken: this.authService.generateAccessToken(userDetails.scopes),
       });
     } catch (err) {
       res.send(err);
```
Comment (SolarFactories) [inline api/src/controllers/AuthenticationController.ts:45]: C: This is not going to send back correct status for errors, sending `200` each time.

File: api/src/controllers/AuthenticationController.ts [hunk 3]
```diff
@@ -1,56 +1,58 @@
...
+  public async signUp(req: Request, res: Response) {
     try {
-      const requestData: UserAttributes = req.body;
+      const data: UserAttributes = req.body;
 
-      // Temp work around
       const userData: UserAttributes = {
-        ...requestData,
-        password: await hashPassword(requestData.password),
+        ...data,
```
Comment (SolarFactories) [inline api/src/controllers/AuthenticationController.ts:54]: S: Potential for passing through untrusted user input.

File: api/src/controllers/AuthenticationController.ts [hunk 4]
```diff
@@ -1,56 +1,58 @@
...
+import AuthService from "../services/AuthenticationService";
 
-export default class AuthController extends ErrorController {
+type GenericClaimStructure = {
+  iss: string;
+  sub: string;
+  aud: string;
+  exp: number;
+  iat: number;
+};
```
Comment (SolarFactories) [inline api/src/controllers/AuthenticationController.ts:11]: C: Duplicated type definition with same name, other within `types/authenticationTypes.ts`.

**Review by SolarFactories — APPROVED** [review]
Most of my comments, are from existing unchanged code, so are able to be resolved in another PR in the future. Once duplicated type definition is de-duplicated can merge.

File: api/src/services/AuthenticationService.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+  public generateAccessToken(scopes: string[]): string {
+    const tokenClaims = this.generateClaims();
+    return jwt.sign(
+      {
+        ...tokenClaims,
+        scope: scopes,
+      },
+      this.getJWTSecretKey(),
+      { algorithm: "HS256" }
```
Comment (SolarFactories) [inline api/src/services/AuthenticationService.ts:49]: W: Use `HS512`.

Comment (Redtigercod4) [reply api/src/services/AuthenticationService.ts]: Good shout. Commit: 66504d03b5cd2997feaa51437addf50dc1d6b874

File: api/src/controllers/AuthenticationController.ts (hunk 4 above)
Comment (Redtigercod4) [reply api/src/controllers/AuthenticationController.ts]: Good spot. Commit: 66504d03b5cd2997feaa51437addf50dc1d6b874

File: api/src/controllers/AuthenticationController.ts (hunk 3 above)
Comment (Redtigercod4) [reply api/src/controllers/AuthenticationController.ts]: Added AJV validation. Commit: 66504d03b5cd2997feaa51437addf50dc1d6b874

File: api/src/controllers/AuthenticationController.ts (hunk 2 above)
Comment (Redtigercod4) [reply api/src/controllers/AuthenticationController.ts:34]: Temp work around at the moment but my next task is to do proper error handling for expressjs. Commit: 66504d03b5cd2997feaa51437addf50dc1d6b874

File: api/src/controllers/AuthenticationController.ts (hunk 1 above)
Comment (Redtigercod4) [reply api/src/controllers/AuthenticationController.ts:12]: Going to fix in the Error Handling PR as need to think on it a bit more.

**Pull request merged**